torchaudio/datasets/datasets.py

```diff
@@ -0,0 +1,419 @@
+import os
+import pickle
+import random
+from functools import partial, reduce
+from warnings import warn
+
+import torch
+
+import torchaudio
+
+
+class Cache:
```

Comment:
you could even make this a function annotations that assumes the input/output relationship is deterministic.